runtime/tests/tests.rs

```diff
@@ -717,6 +726,6 @@ fn contract_call_fails_when_cpay_is_used_for_fee_exchange() {
 				function: contract_call,
 			});
 			let r = Executive::apply_extrinsic(xt);
-			assert_eq!(r, Err(InvalidTransaction::Payment.into()));
+			assert_eq!(r, Err(InvalidTransaction::Custom(UNKNOWN_BUY_FEE_ASSET).into()));
```

Comment:
This one is a rough edge:
 - 'CannotTradeFeeAssetForItSelf' seems wordy
- Could always check this case in 'buy fee asset' and return early knowing it would result in a CPAY payment anyway but more work 🤷‍♂